src/components/waste/WasteCollectionSettings.tsx

```diff
@@ -348,6 +354,16 @@ export const WasteCollectionSettings = ({
     dispatch({ type: WasteSettingsActions.setReminderTime, payload: newTime });
   }, []);
 
+  const getPermission = async () => {
+    const inAppPermission = await getInAppPermission();
+
+    setIsPushPermissionGranted(inAppPermission);
+  };
+
+  useEffect(() => {
+    getPermission();
+  }, [isPushPermissionGranted, showNotificationSettings]);
```

Comment:
`getPermission` is only called in this effect, so it could also be defined only in here, right?
we do this in other places as well to be able to call async in effect